core/nuxeo-connection.js

```diff
@@ -166,7 +176,18 @@ import './nuxeo-element.js';
       // share the connect promise between all instances (one per client)
       this.client._promise = this.client.connect();
 
-      return this.client._promise.then(this._handleConnected.bind(this));
+      return this.client._promise.then(this._handleConnected.bind(this)).catch((error) => {
+        if (error.response.status === 401) {
+          if (this.method === 'form') {
```

Comment:
We're keeping this extra if just to leave room to add clauses for the other authentication methods?

Reply:
and to avoid the redirection when using basic auth

Reply:
But we could get that with a single if (error.response.status === 401 && this.method === 'form')

Reply:
ah! yes, sorry. even had a switch at some point to ease case fall troughs but i guess we'll revisit this anyway when adding oauth2